test/test_tensorboard.py

```diff
@@ -460,13 +460,21 @@ def test_mesh(self):
 def remove_whitespace(string):
     return string.replace(' ', '').replace('\t', '').replace('\n', '')
 
-def read_expected_content(function_ptr):
+def get_expected_file(function_ptr):
     module_id = function_ptr.__class__.__module__
-    test_dir = os.path.dirname(sys.modules[module_id].__file__)
+    test_file = sys.modules[module_id].__file__
+    # Look for the .py file (since __file__ could be pyc).
+    test_file = ".".join(test_file.split('.')[:-1]) + '.py'
+
+    # Use realpath to follow symlinks appropriately.
+    test_dir = os.path.dirname(os.path.realpath(test_file))
     functionName = function_ptr.id().split('.')[-1]
-    expected_file = os.path.join(test_dir,
-                                 "expect",
-                                 'TestTensorBoard.' + functionName + ".expect")
+    return os.path.join(test_dir,
+                        "expect",
+                        'TestTensorBoard.' + functionName + ".expect")
+
+def read_expected_content(function_ptr):
+    expected_file = get_expected_file(function_ptr)
```

Comment:
You don't have to fix this, but this is really weird. I don't know why `test_tensorboard` has another copy of the "find expect file" logic...

Reply:
The logic in common_utils.py is slightly different and it actually uses the `subname` parameter to determine the right file name.

Reply:
That's the story of life with copy pasted code, right :) Maybe if you could file a bug on this and CC the original code author that would be a big help